api4/resolver_channel_member.go

```diff
@@ -65,12 +66,24 @@ func (cm *channelMember) Channel(ctx context.Context) (*channel, error) {
 }
 
 func (cm *channelMember) Roles_(ctx context.Context) ([]*model.Role, error) {
-	c, err := getCtx(ctx)
+	loader, err := getRolesLoader(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	return getGraphQLRoles(c, strings.Fields(cm.Roles))
+	thunk := loader.LoadMany(ctx, dataloader.NewKeysFromStrings(strings.Fields(cm.Roles)))
+	results, errs := thunk()
+	// All errors are the same. We just return the first one.
+	if len(errs) > 0 && errs[0] != nil {
+		return nil, err
+	}
+
+	roles := make([]*model.Role, len(results))
+	for i, res := range results {
+		roles[i] = res.(*model.Role)
```

Comment:
Btw, I think dataloaders are a prime candidate to use generics. The whole thunk API uses `interface{}`. I have talked with the maintainers and they agree as well. Let's see..

Reply:
Cool, looking forward to see proper uses of generics :)